### PR TITLE
Added new preview version for container instances

### PR DESCRIPTION
**Pull request by dobrebogdan**
Creating a new preview api version with a new property "identityAcls" and related classes for azure container instance. Also adding CGProfile API specifications and missing API paths.

### CODE REVIEW

**Comment by dobrebogdan** [comment]
Not using the latest api version as default due to some missing API paths compared to 2024-05 that trigger this error for the Avocado check: [MISSING_APIS_IN_DEFAULT_TAG](https://github.com/Azure/avocado/blob/main/README.md#missing_apis_in_default_tag)

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json [hunk 1]
```diff
@@ -0,0 +1,2666 @@
...
+              "enum": [
+                "Regular",
+                "Spot"
+              ],
+              "x-ms-enum": {
+                "name": "ContainerGroupPriority",
+                "modelAsString": true
+              }
+            },
+            "identityAcls": {
```
Comment (ramoka178) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:1601]: >identityAcls [](http://example.com/codeflow?start=13&length=12)

can it be fully named ? identityAccessControlLevels ?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json [hunk 2]
```diff
@@ -0,0 +1,2666 @@
...
+          "description": "The default access level.",
+          "type": "string",
+          "enum": [
+            "All",
+            "System",
+            "User"
+          ],
+          "x-ms-enum": {
+            "name": "IdentityAccessLevel",
+            "modelAsString": true
```
Comment (ramoka178) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2561]: please add description for each enum on what do they mean

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json [hunk 3]
```diff
@@ -0,0 +1,2666 @@
...
+          }
+        },
+        "acls": {
```
Comment (ramoka178) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2564]: >acls [](http://example.com/codeflow?start=9&length=4)

full name may be ? accessControlLevels ?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json [hunk 4]
```diff
@@ -0,0 +1,2666 @@
...
+          "description": "The access level of the identity.",
+          "type": "string",
+          "enum": [
+            "All",
+            "System",
+            "User"
+          ],
+          "x-ms-enum": {
+            "name": "IdentityAccessLevel",
+            "modelAsString": true
```
Comment (ramoka178) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2590]: please add description for each enum on what do they mean

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json [hunk 5]
```diff
@@ -0,0 +1,2666 @@
...
+          }
+        },
+        "identity": {
+          "description": "An identity.",
+          "type": "string"
+        }
```
Comment (ramoka178) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2596]: is this a guid or any ARM resource id ? 
If its arm resource id, can you add "format":"arm-id" ?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json [hunk 6]
```diff
@@ -0,0 +1,2666 @@
...
+      "description": "The properties for confidential container group",
+      "type": "object",
+      "properties": {
+        "ccePolicy": {
+          "type": "string",
+          "description": "The base64 encoded confidential compute enforcement policy"
+        }
+      }
+    },
+    "IdentityAcls": {
```
Comment (ramoka178) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2547]: >IdentityAcls [](http://example.com/codeflow?start=5&length=12)

so this has 2 levels of acl information, one that is default and one that is for each identity then ?

**Comment by ramoka178** [comment]
> Not using the latest api version as default due to some missing API paths compared to 2024-05 that trigger this error for the Avocado check: [MISSING_APIS_IN_DEFAULT_TAG](https://github.com/Azure/avocado/blob/main/README.md#missing_apis_in_default_tag)

Are these missing API paths not going to be implemented ?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json (hunk 1 above)
Comment (dobrebogdan) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:1601]: I'm afraid not at this point, that name was decided within the team and the feature is already delivered with this name. But I will discuss doing that change in a future version.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json (hunk 3 above)
Comment (dobrebogdan) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2564]: I'm afraid that's not possible at this point, that name was decided within the team and the feature is already delivered with this name. But I will discuss doing that change in a future version.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json (hunk 6 above)
Comment (dobrebogdan) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2547]: Yes, exactly, there is a list of identities with specific access levels and a default access level that will be used for identities not appearing in that list. So you can see it as a default access level that is modified for a list of identities.

**Comment by dobrebogdan** [comment]
> > Not using the latest api version as default due to some missing API paths compared to 2024-05 that trigger this error for the Avocado check: [MISSING_APIS_IN_DEFAULT_TAG](https://github.com/Azure/avocado/blob/main/README.md#missing_apis_in_default_tag)
> 
> Are these missing API paths not going to be implemented ?

I don't think so, they were missing in the previous version as well. I think they were deprecated. That check mentions that it will be a false positive in the case of deprecation.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json (hunk 2 above)
Comment (dobrebogdan) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2561]: There is already a description for this enum a few lines above this tag

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json (hunk 4 above)
Comment (dobrebogdan) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2590]: There is already a description for this enum a few lines above this tag

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json (hunk 5 above)
Comment (dobrebogdan) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:2597]: Yes, I added that, thanks!

**Comment by dobrebogdan** [comment]
> > > Not using the latest api version as default due to some missing API paths compared to 2024-05 that trigger this error for the Avocado check: [MISSING_APIS_IN_DEFAULT_TAG](https://github.com/Azure/avocado/blob/main/README.md#missing_apis_in_default_tag)
> > 
> > 
> > Are these missing API paths not going to be implemented ?
> 
> I don't think so, they were missing in the previous version as well. I think they were deprecated. That check mentions that it will be a false positive in the case of deprecation.

After doing some investigation it seems there is some work to add those paths back on another PR. I contacted the owner of the PR and we will sync them.

**Comment by dobrebogdan** [comment]
> > > > Not using the latest api version as default due to some missing API paths compared to 2024-05 that trigger this error for the Avocado check: [MISSING_APIS_IN_DEFAULT_TAG](https://github.com/Azure/avocado/blob/main/README.md#missing_apis_in_default_tag)
> > > 
> > > 
> > > Are these missing API paths not going to be implemented ?
> > 
> > 
> > I don't think so, they were missing in the previous version as well. I think they were deprecated. That check mentions that it will be a false positive in the case of deprecation.
> 
> After doing some investigation it seems there is some work to add those paths back on another PR. I contacted the owner of the PR and we will sync them.

The PRs are synched, nothing is missing now.

**Comment by raosuhas** [comment]
please fix the newly introduced lintdiff warnings from here : https://github.com/Azure/azure-rest-api-specs/pull/31223/checks?check_run_id=32598293246

**Comment by dobrebogdan** [comment]
> please fix the newly introduced lintdiff warnings from here : https://github.com/Azure/azure-rest-api-specs/pull/31223/checks?check_run_id=32598293246

Are any of these warnings a blocker or a serious issue? This PR is important for Ignite which is soon, so we would prefer to merge if they can be ignored. What do you think?

**Comment by raosuhas** [comment]
> > please fix the newly introduced lintdiff warnings from here : https://github.com/Azure/azure-rest-api-specs/pull/31223/checks?check_run_id=32598293246
> 
> Are any of these warnings a blocker or a serious issue? This PR is important for Ignite which is soon, so we would prefer to merge if they can be ignored. What do you think?

you can ignore the ones that say : ⚠️ [AvoidNestedProperties](https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-nested-properties.md) 

FOr the [EnumInsteadOfBoolean](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md) I will leave it to your team to review these and change since this is a strong recommendation. 

For the others they must be fixed. These are straightforward fixes and should not really take you that long to fix.

**Comment by dobrebogdan** [comment]
> > > please fix the newly introduced lintdiff warnings from here : https://github.com/Azure/azure-rest-api-specs/pull/31223/checks?check_run_id=32598293246
> > 
> > 
> > Are any of these warnings a blocker or a serious issue? This PR is important for Ignite which is soon, so we would prefer to merge if they can be ignored. What do you think?
> 
> you can ignore the ones that say : ⚠️ [AvoidNestedProperties](https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-nested-properties.md)
> 
> FOr the [EnumInsteadOfBoolean](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md) I will leave it to your team to review these and change since this is a strong recommendation.
> 
> For the others they must be fixed. These are straightforward fixes and should not really take you that long to fix.

All lint diff warnings except for the 2 that were acceptable were fixed.

Among other things, the SubscriptionIdParameter in the file was removed and replaced with the one from common-types. However, that one has something which was missing in the parameter in the file: "format": "uuid".

**!!!This is what triggered the breaking change check, but it's not a "real" breaking change, the usage is not changed and that parameter was always supposed to be a uuid.**

**Comment by razvanbadea-msft** [comment]
>             "maxBatchPercent": {

you can add the limits for minimum and maximum of 100 if this is 0-100% range

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:3605 in c4654f1. [](commit_id = c4654f1b101e87937bee1ead9c32c4a52d8635ee, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>             "inPlaceUpdate": {

you can use in this case an enum like rollingUpdateType with values like [InPlace, Recreate]

Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:3619 in c4654f1. [](commit_id = c4654f1b101e87937bee1ead9c32c4a52d8635ee, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>               "enum": [

why the values of this enum is all lowecase and not using the PascalCase as in the provided link and like shareAccessType?

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:3816 in c4654f1. [](commit_id = c4654f1b101e87937bee1ead9c32c4a52d8635ee, deletion_comment = False)

**Comment by shivg7795** [comment]
> > ```
> >           "enum": [
> > ```
> 
> why the values of this enum is all lowecase and not using the PascalCase as in the provided link and like shareAccessType?
> 
> Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:3816 in [c4654f1](https://github.com/Azure/azure-rest-api-specs/commit/c4654f1b101e87937bee1ead9c32c4a52d8635ee). [](commit_id = [c4654f1](https://github.com/Azure/azure-rest-api-specs/commit/c4654f1b101e87937bee1ead9c32c4a52d8635ee), deletion_comment = False)

this change is not related to this PR. We can definitely address it in upcoming version but can we please ignore it if it is not blocking.

**Comment by razvanbadea-msft** [comment]
> > > ```
> > >           "enum": [
> > > ```
> > 
> > 
> > why the values of this enum is all lowecase and not using the PascalCase as in the provided link and like shareAccessType?
> > Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:3816 in [c4654f1](https://github.com/Azure/azure-rest-api-specs/commit/c4654f1b101e87937bee1ead9c32c4a52d8635ee). [](commit_id = [c4654f1](https://github.com/Azure/azure-rest-api-specs/commit/c4654f1b101e87937bee1ead9c32c4a52d8635ee), deletion_comment = False)
> 
> this change is not related to this PR. We can definitely address it in upcoming version but can we please ignore it if it is not blocking.

this enum is added in this PR so it is related

**Comment by razvanbadea-msft** [comment]
>               "description": "learn more at: https://learn.microsoft.com/en-us/rest/api/storagerp/file-shares/create?tabs=HTTP#shareaccesstier",

Could you update this description a bit? This are used to generate comments & documentation, so they're used by customers to discover what these properties mean.

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:3815 in 57781eb. [](commit_id = 57781ebe0ecce1898909948a6ca598d41c810bcb, deletion_comment = False)

**Comment by shivg7795** [comment]
> > ```
> >           "description": "learn more at: https://learn.microsoft.com/en-us/rest/api/storagerp/file-shares/create?tabs=HTTP#shareaccesstier",
> > ```
> 
> Could you update this description a bit? This are used to generate comments & documentation, so they're used by customers to discover what these properties mean.
> 
> Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-11-01-preview/containerInstance.json:3815 in [57781eb](https://github.com/Azure/azure-rest-api-specs/commit/57781ebe0ecce1898909948a6ca598d41c810bcb). [](commit_id = [57781eb](https://github.com/Azure/azure-rest-api-specs/commit/57781ebe0ecce1898909948a6ca598d41c810bcb), deletion_comment = False)

addressed all the comments

**Comment by shivg7795** [comment]
@razvanbadea-msft this swagger version is critical for upcoming Ignite event. Hence please suggest if there are any other important/blocking changes, I can address those asap. The swagger needs to be published at priority before the Ignite event.

**Review by razvanbadea-msft — APPROVED** [review]

**Pull request merged**